### ISSUE TITLE
EvilLoggerTests.testDeprecatedSettings fails assertion

### ISSUE THREAD

**Issue by talevy** [elastic/elasticsearch#25680]
looks like a consistent behavior where-in this test fails on Java 9 in CI

console logs:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+java9-periodic/3193/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+java9-periodic/3196/console
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.x+java9-periodic/3198/console

```
FAILURE 0.16s J0 | EvilLoggerTests.testDeprecatedSettings <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <1>
   >      but: was <0>
   > 	at __randomizedtesting.SeedInfo.seed([FA22CEC1F9E81B67:9A5B351BD2070DE8]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.common.logging.EvilLoggerTests.testDeprecatedSettings(EvilLoggerTests.java:271)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   > 	at java.base/java.lang.Thread.run(Thread.java:844)
  2> NOTE: leaving temporary files on disk at: /var/lib/jenkins/workspace/elastic+elasticsearch+5.x+java9-periodic/qa/evil-tests/build/testrun/test/J0/temp/org.elasticsearch.common.logging.EvilLoggerTests_FA22CEC1F9E81B67-001
  2> NOTE: test params are: codec=Asserting(Lucene62): {}, docValues:{}, maxPointsInLeafNode=699, maxMBSortInHeap=6.526490608214228, sim=RandomSimilarity(queryNorm=false,coord=no): {}, locale=dz, timezone=Asia/Kamchatka
  2> NOTE: Linux 4.8.0-53-generic amd64/Oracle Corporation 9 (64-bit)/cpus=8,threads=1,free=333207680,total=536870912
  2> NOTE: All tests run in this JVM: [PluginSecurityTests, ESPolicyUnitTests, EvilSecurityTests, EvilLoggerTests]
Completed [14/14] on J0 in 4.81s, 10 tests, 1 failure <<< FAILURES!
```

reproduce with (I cannot reproduce):

```
gradle :qa:evil-tests:test -Dtests.seed=6F4D930464F6F74A -Dtests.class=org.elasticsearch.common.logging.EvilLoggerTests -Dtests.method="testDeprecatedSettings" -Dtests.security.manager=false -Dtests.locale=en-IE -Dtests.timezone=Universal
```

**Comment by talevy** [comment]
happened again... not in java9

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+oracle-periodic/157/console

**Comment by andy-elastic** [comment]
Another one in 5.6 https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+5.6+aggressive-opts/49/console

I'll awaitsfix this test since it's happened a handful of times this week

**Comment by jasontedor** [comment]
This one is so incredibly puzzling. I can get it to reproduce spuriously, but only on 5.6. Still no idea what is going on. 😐

**Comment by jasontedor** [comment]
I think I found the issue, it's from interaction with another test.

**Comment by jasontedor** [comment]
And it only happens on 5.6 because the interaction is from deprecated use of `path.scripts`, which does not exist in 6.0, 6.x, and master.

**Comment by jasontedor** [comment]
I have this one nailed down. It uncovered an issue with #20575 and #24076, I will open a pull request for this first. Then I will address this test failure.

**Comment by talevy** [comment]
sweet! so I'll drop the `6.1.0` tag

**Comment by jasontedor** [comment]
Step 1: #26209

**Comment by jasontedor** [comment]
Step 2: #26210

**Comment by jasontedor** [comment]
Step 3: b1f6131b30b5c3aff24b243f4fb1b706385bb8e8

**Comment by jasontedor** [comment]
Closed by b1f6131b30b5c3aff24b243f4fb1b706385bb8e8
